applications/box.md

```diff
@@ -0,0 +1,213 @@
+
+Below we provide an **example roadmap**. In the descriptions, it should be clear how your project is related to IPFS/Crust. We *recommend* that the scope of the work can fit within a three-month period and that teams structure their roadmap as 1 milestone ≈ 1 month. 
+
+For each milestone,
+
+* make sure to include a specification of your software. _Treat it as a contract_; the level of detail must be enough to later verify that the software meets the specification.
+* include the amount of funding requested _per milestone_.
+* include documentation (tutorials, API specifications, architecture diagrams, whatever is appropriate) in each milestone. This ensures that the code can be widely used by the community.
+* provide a test suite, comprising unit and integration tests, along with a guide on how to set up and run them.
+* commit to providing Dockerfiles for the delivery of your project. 
+* indicate milestone duration as well as number of full-time employees working on each milestone, and include the approximate number of days along with the cost per day.
+* _Deliverables 0a-0d are mandatory_ and shall not be removed, unless you explicitly specify a reason within the PR's `Additional Notes` section (e.g. Milestone X is research oriented and as such there is no code to test).
+
+> If any of your deliverables is based on somebody else's work, make sure you work and publish _under the terms of the license_ of the respective project and that you **highlight this fact in your milestone documentation** and in the source code if applicable! **Teams that submit others' work without attributing it will be immediately terminated.**
+
+### Overview
+* **Total Estimated Duration:** 3 months
+* **Full-Time Equivalent (FTE):**  3 months as we are working on it full-time
+* **Total Costs:** $28K
+
+### Milestone 1 — Implement Crust Order Modules 
+* **Estimated Duration:** 1 month
+* **FTE:**  2
+* **Costs:** 10,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 / MIT / Unlicense |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Crust nodes. Once the node is up, it will be possible to send test transactions that will show how the new functionality works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. | 
+| 1. | File transfer module from applications to box | We will create an IPFS+Libp2p module that transfers the file from application to the box backend |  
+| 2. | File transfer functionality from box to applications | We will create an IPFS+Libp2p module that transfers the file from box to the front-end application on request | 
+| 3. | Reading photos in photos app from box | We will connect photos app to store and read photos from box backend |  
```

Comment:
One advantage of Crust is that user files can be saved by IPFS nodes decentralized and distributed around the world. The availability of files in IPFS is extremely high. What I want to know is how the box guarantees the user’s download speed and the reliability of the box itself?

Reply:
This is exactly the advantage that we hope to utilize in this proposal. When sharing between users there is no way other than having them pinned on a service like crust, however while the photo is not shared with anyone and user is within the local network(eg at home) we leave this choice to user if they want it pinned or not as we want to enable them run it at zero cost if they choose to, and we leave it to user if they want to disconnect a box from network for any privacy reasons.

box serves the following purposes to complement the current ecosystem with a familiar decentralized alternative for the widely popular client-server model:

1- to provide APIs that front-end can communicate with it regardless of platform without running a full ipfs node. This makes app development do-able by front-end developers without needing to understand back-end and/or worrying about running ipfs on the target device. 
2- user has a copy of the files locally as well as keeping them on pinning services (we leave this choice to user as we want to enable them run it at zero cost if hey choose to) 
3- It provides a pre-setup ipfs node for user so that non-tech-savvy users can still benefit from the ipfs

Reply:
This is updated in response to: https://github.com/crustio/Crust-Grants-Program/pull/39/files#r669248392

```suggestion
| 3. | Reading photos in photos app from box | We will connect photos app to store and read photos from box backend. We do not provide IPFS gateway service as it is not required. A box app knows the peerIds of it's host box nodes (through scanning a QR code while initiating the app) and dials files protocol directly on a libp2p swarm that is connecting all destination boxes. |  
```